views/doctorate/cdd/list.py

```diff
@@ -0,0 +1,108 @@
+from django.core.exceptions import NON_FIELD_ERRORS
+from django.db.models import QuerySet
+from django.shortcuts import render
+from django.utils.translation import gettext as _
+from django.views.generic import ListView
+
+from admission.contrib.models import DoctorateAdmission
+from admission.ddd.projet_doctoral.validation.commands import FiltrerDemandesQuery
+from admission.forms.doctorate.cdd.filter import FilterForm
+from admission.auth.mixins import CddRequiredMixin
+from base.utils.htmx import HtmxMixin
+from infrastructure.messages_bus import message_bus_instance
+
+
+class CddDoctorateAdmissionList(LoginRequiredMixin, CddRequiredMixin, HtmxMixin, ListView):
+    model = DoctorateAdmission
+    raise_exception = True
+
+    template_name = 'admission/doctorate/cdd/list.html'
+    htmx_template_name = 'admission/doctorate/cdd/list_block.html'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.form = None
+
+    @staticmethod
+    def htmx_render_form_errors(request, form):
+        """Returns an HttpResponse containing the errors of the form grouped by field."""
+        field_errors = form.errors
+        non_field_errors = field_errors.pop(NON_FIELD_ERRORS, [])
+        formatted_errors = [
+            (form.fields.get(field_name).label, errors) for field_name, errors in field_errors.items()
+        ]
+        if non_field_errors:
+            formatted_errors.append((_('General'), non_field_errors))
+
+        return render(
+            status=400,
+            request=request,
+            template_name='admission/includes/form_errors.html',
+            context={
+                'errors': formatted_errors,
+            },
+        )
+
+    def get(self, request, *args, **kwargs):
+        self.form = FilterForm(user=self.request.user, data=self.request.GET or None)
+
+        if not self.form.is_valid() and self.request.htmx:
+            return self.htmx_render_form_errors(request, self.form)
+
+        return super().get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['filter_form'] = self.form
+        return context
+
+    def get_paginate_by(self, queryset):
+        return self.request.GET.get('page_size')
+
+    def get_queryset(self) -> QuerySet:
+        if self.form.is_bound and self.form.is_valid():
+
+            filters = self.form.cleaned_data
+
+            filters.pop('page_size', None)
+
+            # Order the queryset
+            ordering_field = self.request.GET.get('o')
+            if ordering_field:
+                filters['tri_inverse'] = ordering_field[0] == '-'
+                filters['champ_tri'] = ordering_field.lstrip('-')
+
+            return message_bus_instance.invoke(
+                FiltrerDemandesQuery(
+                    **filters,
+                )
+            )
+
+        else:
+            return []
```

Comment:
Préférer la méthode du "early return", deux avantages : 
 - tu te débarrasse des cas "extremes" au début
 - ton indentation est moindre
 - le véritable corps/métier de ta fonction n'est pas dans un if
```suggestion
        if not self.form.is_bound or not self.form.is_valid():
            return []

        filters = self.form.cleaned_data

        filters.pop('page_size', None)

        # Order the queryset
        ordering_field = self.request.GET.get('o')
        if ordering_field:
            filters['tri_inverse'] = ordering_field[0] == '-'
            filters['champ_tri'] = ordering_field.lstrip('-')

        return message_bus_instance.invoke(FiltrerDemandesQuery(**filters))
```